### PR TITLE
Basic Plugin API v 1.2 

### PR DESCRIPTION
**Pull request by Tmeister**
# Pull Request

## What changed?

- Adds the ability to respond to the Plugins API version 1.2 `plugin_information ` and a basic `query_plugins.`
- I add factories to `SyncPlugins` and `Plugins` to get some data to test on.
- New `Plugin Resource` and `Plugin Collection` to handle the response format in one place

**This PR contains #65** 

## Why did it change?

To handle the plugin 1.2 API version requests

## Did you fix any specific issues?

NA

## CERTIFICATION

By opening this pull request, I do agree to abide by
the [CODE OF CONDUCT](https://github.com/aspirepress/.github/CODE_OF_CONDUCT.md) and be bound by the terms
of the [Contribution Guidelines](https://github.com/aspirepress/.github/CONTRIBUTING.md) in effect on the date and time
of my contribution as proven by the
revision information in GitHub. I also agree that any previous contributions shall be deemed subject to the terms of the
version in effect on the date and time of this pull request, or any future revisions for pull requests I may submit.
Further, I certify that this work is my own, is original, does not violate the intellectual property of any other person
or entity, and I am not violating any license agreements or contracts I have with any person or entity. Finally, I agree
that this code may be licensed under any license deemed appropraite by AspirePress, including but not
limited to open source, closed source, proprietary or custom licenses, and that such license terms neither violate my
rights or my copyright to this code.

### CODE REVIEW

**Comment by chuckadams** [comment]
Going to go over it with a more proper review soon, but overall this looks pretty good.  Being written in parallel with the things I was doing with laravel-data, this is sort of a parallel evolution, using Laravel's resource abstractions instead of Data classes.  While I'm not big on the design of Laravel's Resource _Controllers_, resource classes themselves seem just fine.  The two will have to meet in the middle sometime, but hopefully we get the best of both out of them.

File: app/Http/Controllers/API/WpOrg/Plugins/Plugin_1_2_Controller.php [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+            return response()->json(['error' => 'Plugin not found'], 404);
+        }
+
+        return response()->json(new PluginResource($plugin));
+    }
+
+    private function queryPlugins(Request $request): JsonResponse
+    {
+        $page = max(1, (int) $request->query('page', '1'));
+        $perPage = (int) $request->query('per_page', '23');
```
Comment (chuckadams) [inline app/Http/Controllers/API/WpOrg/Plugins/Plugin_1_2_Controller.php:50]: should default to 24 per page, that's the default for themes, and a quick query to upstream shows it's the case for plugins too.

File: phpunit.xml [hunk 2]
```diff
@@ -22,8 +22,12 @@
         <env name="APP_MAINTENANCE_DRIVER" value="file"/>
         <env name="BCRYPT_ROUNDS" value="4"/>
         <env name="CACHE_STORE" value="array"/>
-        <!-- <env name="DB_CONNECTION" value="sqlite"/> -->
-        <!-- <env name="DB_DATABASE" value=":memory:"/> -->
+        <env name="DB_CONNECTION" value="pgsql"/>
+        <env name="DB_HOST" value="localhost"/>
```
Comment (chuckadams) [inline phpunit.xml:26]: Should be `postgres`, not localhost, since tests are intended to run in the container.

**Review by chuckadams — CHANGES_REQUESTED** [review]

File: app/Http/Controllers/API/WpOrg/Plugins/Plugin_1_2_Controller.php (hunk 1 above)
Comment (Tmeister) [reply app/Http/Controllers/API/WpOrg/Plugins/Plugin_1_2_Controller.php]: Fixed, changed to 24

File: phpunit.xml (hunk 2 above)
Comment (Tmeister) [reply phpunit.xml]: DB_HOST value now is `postgres`

**Review by chuckadams — APPROVED** [review]

**Pull request merged**